src/v/kafka/server/group.cc

```diff
@@ -2945,21 +2872,22 @@ group::do_commit(kafka::group_id group_id, model::producer_identity pid) {
     // problem, because client got ok for commit_request (see
     // tx_gateway_frontend). So redpanda will eventually finish commit and
     // complete write for both this events.
+
     model::record_batch_reader::data_t batches;
     batches.reserve(2);
 
     cluster::simple_batch_builder store_offset_builder(
       model::record_batch_type::raft_data, model::offset(0));
-    for (const auto& [tp, metadata] : ongoing_it->second.offsets) {
+    for (const auto& [tp, pending_offset] : ongoing_tx.offsets) {
         update_store_offset_builder(
           store_offset_builder,
           tp.topic,
           tp.partition,
-          metadata.offset,
-          metadata.committed_leader_epoch,
-          metadata.metadata,
-          metadata.commit_timestamp,
-          metadata.expiry_timestamp);
+          pending_offset.offset_metadata.offset,
+          kafka::leader_epoch(pending_offset.offset_metadata.leader_epoch),
+          pending_offset.offset_metadata.metadata.value_or(""),
+          model::timestamp::now(),
+          std::nullopt);
```

Comment:
should these be actual timestamps?

Reply:
this was previously set to the same value when the offset was stored, this is in align to what Kafka does.